src/libcrun/container.c

```diff
@@ -3706,6 +3773,160 @@ libcrun_container_update_from_values (libcrun_context_t *context, const char *id
+
+  (*capabilities)[index] = NULL; // Terminate the array with NULL
+  populate_array_field (&(info->linux.capabilities), *capabilities, *num_capabilities);
+}
+
+void
+retrieve_mount_options (struct features_info_s **info)
+{
+  // Retrieve mount options from wordlist
+  const struct propagation_flags_s *mount_options_list = get_mount_flags_from_wordlist ();
+
+  // Calculate the number of mount options
+  size_t num_mount_options = 0;
+  while (mount_options_list[num_mount_options].name != NULL)
+    {
+      num_mount_options++;
+    }
+
+  // Allocate memory for mount options in info struct
+  (*info)->mount_options = xmalloc0 (num_mount_options * sizeof (char *));
+
+  // Copy mount options to info struct
+  for (size_t i = 0; i < num_mount_options; i++)
+    {
+      (*info)->mount_options[i] = xstrdup (mount_options_list[i].name);
+    }
+}
+
+int
+libcrun_container_get_features (struct features_info_s **info)
+{
+  // Allocate memory for the features_info_s structure
+  *info = xmalloc0 (sizeof (struct features_info_s));
+  // Hardcoded feature information
+  (*info)->oci_version_min = xstrdup ("1.0.0");
+  (*info)->oci_version_max = xstrdup ("1.1.0-rc.3");
+
+  // Populate hooks
+  size_t num_hooks = sizeof (hooks) / sizeof (hooks[0]);
+  populate_array_field (&((*info)->hooks), hooks, num_hooks);
+
+  // Populate mount_options
+  retrieve_mount_options (info);
+
+  // Populate namespaces
+  size_t num_namspaces = sizeof (namespaces) / sizeof (namespaces[0]);
+  populate_array_field (&((*info)->linux.namespaces), namespaces, num_namspaces);
+
+  // Populate capabilities
+  size_t num_capabilities = 0;
+  const char **capabilities = NULL;
+  populate_capabilities (*info, &capabilities, &num_capabilities);
+
+  // Hardcode the values for cgroup
+  (*info)->linux.cgroup.v1 = true;
+  (*info)->linux.cgroup.v2 = true;
+#ifdef HAVE_SYSTEMD
+  (*info)->linux.cgroup.systemd = true;
+  (*info)->linux.cgroup.systemdUser = true;
+#endif
+
+  // Put seccomp values
+  (*info)->linux.seccomp.enabled = true;
+  // Populate actions
+  size_t num_actions = sizeof (actions) / sizeof (actions[0]);
+  populate_array_field (&((*info)->linux.seccomp.actions), actions, num_actions);
+  // Populate operators
+  size_t num_operators = sizeof (operators) / sizeof (operators[0]);
+  populate_array_field (&((*info)->linux.seccomp.operators), operators, num_operators);
+  // Populate archs
+  size_t num_archs = sizeof (archs) / sizeof (archs[0]);
+  populate_array_field (&((*info)->linux.seccomp.archs), archs, num_archs);
+
+  // Put values for apparmor and selinux
+  (*info)->linux.apparmor.enabled = true;
+  (*info)->linux.selinux.enabled = true;
+
+  // Populate the values for annotations
+#ifdef HAVE_SECCOMP
+  {
+    const struct scmp_version *version = seccomp_version ();
+    struct annotations_info_s annotations;
+
+    int size = snprintf (NULL, 0, "%u.%u.%u", version->major, version->minor, version->micro) + 1;
+    char *version_string = xmalloc0 (size);
```

Comment:
declarations at the beginning of the function

Reply:
I think this block will be executed only when ` HAVE_SECCOMP` is true, so I thought of declaring it there.

Reply:
you are right, better here